dirwatcher.py

```diff
@@ -1,56 +1,92 @@
-    ))
-
-    # Keys are the actual filename and the values are where to begin searching.
-
-    # Look at directory and get a list of files from it.
-    # Add these to dictonary if not already present, and log it as a new file.
-    
-    # Look through your "watching_files" dictonary
-    # and compare that to a list of files that are in the dictonary.
-    
-    # If the file is not in your dictonary anymore you have
-    # to log the file and remove it from your dictonary
-
-    # Iterate through dictonary, open each file at the last line you read from, 
-    # Start reading from that point looking for any "magic" text
-    
-    # Update the last position that you read from in the dictionary  
-    
-    while True: 
-        try:
-            logger.info('Inside Watch Loop')
-            time.sleep(args.interval)
-        except KeyboardInterrupt:
-            break
+    """Watches given directory and reports when files matching the
+    given extension are added or removed.  Calls find_magic to search
+    present files for a given magic word"""
+    global files_found
+    global magic_word_position
+    logger.info('Watching directory: {}, File Ext: {}, Polling Interval: {}, '
+                'Magic Text: {}'.format(
+                    args.path, args.ext, args.interval, args.magic
+                ))
+    directory = os.path.abspath(args.path)
+    files_in_directory = os.listdir(directory)
+    for file in files_in_directory:
+        if file.endswith(args.ext) and file not in files_found:
+            logger.info('New file: {} found in {}'.format(file, args.path))
+            files_found.append(file)
+            magic_word_position[file] = 0
+    for file in files_found:
+        if file not in files_in_directory:
+            logger.info('File: {} removed from {}'.format(file, args.path))
+            files_found.remove(file)
+            del magic_word_position[file]
+    for file in files_found:
+        find_magic(file, args.magic, directory)
+
+
+def signal_handler(sig_num, frame):
+    """Looks for signals SIGINT and SIGTERM and toggles the exit_flag"""
+    global exit_flag
+    # log the associated signal name (the python3 way)
+    # logger.warn('Received ' + signal.Signals(sig_num).name)
+    # log the signal name (the python2 way)
+    signames = dict((k, v) for v, k in reversed(sorted(
+        signal.__dict__.items()))
+        if v.startswith('SIG') and not v.startswith('SIG_'))
+    logger.warning('Received signal: ' + signames[sig_num])
+    if sig_num == signal.SIGINT or signal.SIGTERM:
+        exit_flag = True
+
+
+def find_magic(filename, magic_word, directory):
+    """Search for the magic word in the filename line by line and
+    keep track of the last line searched"""
+    global magic_word_position
+    with open(directory + '/' + filename) as f:
+        for i, line in enumerate(f.readlines(), 1):
+            if magic_word in line and i > magic_word_position[filename]:
```

Comment:
I would reverse the order of this statement
`magic_word in line and i > magic_word_position[filename]:`
to be this instead
`i > magic_word_position[filename] and magic_word in line`
because in the first form, you are always testing for the magic string, even when you have not yet reached the line number you want to be checking after-- it is inefficient